lib/post_revisor.rb

```diff
@@ -137,6 +156,58 @@ def self.track_and_revise(topic_changes, field, attribute)
     end
   end
 
+  def self.create_small_action_for_category_change(topic_changes:, old_category:, new_category:)
+    return if !SiteSetting.create_post_for_category_and_tag_changes
+
+    topic_changes.topic.add_moderator_post(
+      topic_changes.user,
+      I18n.t(
+        "topic_category_changed",
+        from: category_name_raw(old_category),
+        to: category_name_raw(new_category),
```

Comment:
I think if the category name changes, the text being stored in the post will no longer be accurate? Perhaps we should explore ways to store the category ids and then generate the text on the client side?

Reply:
Also the text here end up defaulting to the site's default locale so this way will not work well for sites that allow users to configure their preferred locale.